pkgs/tools/system/uefitool/new-engine.nix

```
@@ -0,0 +1,37 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  qtbase,
+  cmake,
+  wrapQtAppsHook,
+  zip,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "uefitool";
+  version = "A71";
+
+  src = fetchFromGitHub {
+    sha256 = "sha256-NRlrKm5+eED6oyvFRSEhn0EUbMsPJtuFAyv3vgY/IUI=";
+    owner = "LongSoft";
+    repo = pname;
+    rev = version;
+  };
+
+  buildInputs = [ qtbase ];
+  nativeBuildInputs = [
+    cmake
+    zip
+    wrapQtAppsHook
+  ];
+  patches = lib.optionals stdenv.isDarwin [ ./bundle-destination.patch ];
+
+  meta = with lib; {
+    description = "UEFI firmware image viewer and editor";
+    homepage = "https://github.com/LongSoft/uefitool";
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ athre0z ];
+    platforms = platforms.linux ++ platforms.darwin;
+  };
```

```suggestion
  meta = {
    description = "UEFI firmware image viewer and editor";
    homepage = "https://github.com/LongSoft/uefitool";
    license = lib.licenses.bsd2;
    maintainers = with lib.maintainers; [ athre0z ];
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
    mainProgram = "UEFITool"
  };
```
The usage of `with lib;` outside of small scopes has been considered as an anti-pattern fore some time now in `nixpkgs`. See #208242